win_guest_debugging_tool: Optimize and simplify script

6-dehan opened the pull request:
The test script for the Windows guest debugging tool was
  significantly refactored to improve its structure,
  eliminate redundancy, and increase maintainability.

  Key logical improvements include:
   - The complex three-class inheritance structure was consolidated into a single, self-contained WinDebugToolTest class, simplifying the overall architecture.
   - A centralized helper method, _run_script_and_get_paths, was created to handle all PowerShell script executions and output parsing. This removes significant code duplication that was present in each test variant.
   - Individual test methods were streamlined to focus purely on their specific validation logic, delegating common tasks like script execution and session management to helper functions.
   - VM session management was made more robust to ensure sessions are properly created, reused, and cleaned up.

ID: 3274
Signed-off-by: Dehan Meng <demeng@redhat.com>

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Tests**
  * Consolidated Windows guest debug-tool checks into one self-contained test with a single execute flow, centralized variant dispatch, unified session/temp-workspace handling, discrete validation/packaging checks, and added configurable script timeouts (including a longer Win2016 timeout) plus first-boot log collection.

* **Bug Fixes**
  * Quieter process termination to avoid spurious failures; improved assertions and error reporting.

* **Chores**
  * Added commands to disable devices and force file-copy overwrites; added first-boot file creation/check steps.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

6-dehan committed:
win_guest_debugging_tool: Optimize and simplify script

  The test script for the Windows guest debugging tool was
  significantly refactored to improve its structure,
  eliminate redundancy, and increase maintainability.

  Key logical improvements include:
   - The complex three-class inheritance structure was
     consolidated into a single, self-contained WinDebugToolTest
     class, simplifying the overall architecture.
   - A centralized helper method, _run_script_and_get_paths,
     was created to handle all PowerShell script executions and
     output parsing. This removes significant code duplication
     that was present in each test variant.
   - Individual test methods were streamlined to focus purely on
     their specific validation logic, delegating common tasks like
     script execution and session management to helper functions.
   - VM session management was made more robust to ensure sessions
     are properly created, reused, and cleaned up.

Signed-off-by: Dehan Meng <demeng@redhat.com>

leidwang commented on qemu/tests/cfg/win_guest_debugging_tool.cfg:
```diff
@@ -13,10 +13,15 @@
+    Win2016:
+        script_execution_timeout = 1200
+        cmd_disable_ie_esc_admin = powershell.exe -Command "Set-ItemProperty -Path 'HKLM:\SOFTWARE\Microsoft\Active Setup\Installed Components\{A509B1A7-37EF-4b3f-8CFC-4F3A74704073}' -Name 'IsInstalled' -Value 0"
+        cmd_disable_ie_esc_user = powershell.exe -Command "Set-ItemProperty -Path 'HKLM:\SOFTWARE\Microsoft\Active Setup\Installed Components\{A509B1A8-37EF-4b3f-8CFC-4F3A74704073}' -Name 'IsInstalled' -Value 0"
```
As we dicsussed offline, it would be better to break down current commit, it would be more clear.

6-dehan replied:
Yeah, I'll split another one, as for this one, just keep it here this time. thanks

6-dehan committed:
win_guest_debugging_tool.py: Add new case for MTV Firstboot log

New requirement from customer that collect MTV Firstboot log.txt
by our script 'CollectSystemInfo.ps1'.
Test steps are:
1. Check the 'C:\Program Files\Guestfs\Firstboot\log.txt' existed
or not, if not, will create a corresponding exact file.
2. Run script
3. Check the file existed or not under generated directory.

Signed-off-by: Dehan Meng <demeng@redhat.com>

leidwang reviewed (approved):
LGTM.

6-dehan committed:
win_guest_debugging_tool: Add IE ESC disable logic

Add IE ESC disable logic for Win2016.

Signed-off-by: Dehan Meng <demeng@redhat.com>
